src/utils/browser-connection-error-hints.ts

```diff
@@ -0,0 +1,39 @@
+import { flattenDeep, uniq } from 'lodash';
+import humanizeDuration from 'humanize-duration';
+import { BrowserSetOptions } from '../runner/interfaces';
+import BrowserConnection from '../browser/connection';
+import { LOCAL_BROWSER_INIT_TIMEOUT, REMOTE_BROWSER_INIT_TIMEOUT } from './browser-connection-timeouts';
+import { CONNECTION_ERROR_HINTS } from '../errors/types';
+import TEMPLATES from '../errors/runtime/templates';
+import renderTemplate from './render-template';
+
+// NOTE: hint about too high concurrency factor will be added to the error after exceeding this value
+const CONCURRENCY_FACTOR_UPPERBOUND = 3;
+
+function getUsedTimeoutMsg (browserInitTimeout?: number): string {
+    if (browserInitTimeout) {
+        const browserInitTimeoutStr = humanizeDuration(browserInitTimeout);
+
+        return `${browserInitTimeoutStr} for all browsers`;
+    }
+
+    const localInitTimeoutStr  = humanizeDuration(LOCAL_BROWSER_INIT_TIMEOUT);
+    const remoteInitTimeoutStr = humanizeDuration(REMOTE_BROWSER_INIT_TIMEOUT);
+
+    return `${localInitTimeoutStr} for local browsers and ${remoteInitTimeoutStr} for remotes`;
+}
+
+export function createHints (connections: BrowserConnection[], opts: BrowserSetOptions): string[] {
```

Comment:
We usually use the `create` for complex object. Since hints just a string, it's better to use the `getHints` naming.

Change method signature to `export default...` and rename file as `get-hints` and move to the `src/browser/connection/` folder.